File: Groupe2/TP1/src/calculs.py
# calcul.py

def main():
    num1 = int(input("Entrez le premier nombre : "))
    num2 = int(input("Entrez le second nombre : "))
    op = input("Entrez l'opérateur (+, -, *, /, %, &, |, ~) : ")

    match op:
        case '+':
            # Addition
            print(f"Résultat : {num1 + num2}")
        case '-':
            # Soustraction
            print(f"Résultat : {num1 - num2}")
        case '*':
            # Multiplication
            print(f"Résultat : {num1 * num2}")
        case '/':
            # Division avec gestion de la division par zéro
            if num2 != 0:
                print(f"Résultat : {num1 / num2}")
            else:
                print("Erreur : Division par zéro.")
        case '%':
            # Modulo avec gestion de la division par zéro
            if num2 != 0:
                print(f"Résultat : {num1 % num2}")
            else:
                print("Erreur : Division par zéro.")
        case '&':
            # Opérateur AND bit à bit
            print(f"Résultat : {num1 & num2}")
        case '|':
            # Opérateur OR bit à bit
            print(f"Résultat : {num1 | num2}")
        case '~':
            # Opérateur NOT bit à bit
            print(f"Résultat : {~num1}")
        case _:
            # Cas par défaut si l'opérateur n'est pas reconnu
            print("Opérateur non reconnu.")

# Cette condition vérifie si le script est exécuté directement plutôt qu'importé comme module
if __name__ == "__main__":
    main()

File: Groupe2/TP1/src/calculs.c
// calculs.c

#include <stdio.h>
#include <limits.h>

int main() {
    int num1, num2;
    char op;

    // Demande d'entrer les valeurs
    printf("Entrez le premier nombre : ");
    scanf("%d", &num1);
    printf("Entrez le second nombre : ");
    scanf("%d", &num2);
    printf("Entrez l'operateur (+, -, *, /, %, &, |, ~) : ");
    scanf(" %c", &op);

    switch (op) {
        case '+':
            // Addition
            printf("Resultat : %d\n", num1 + num2);
            break;
        case '-':
            // Soustraction
            printf("Resultat : %d\n", num1 - num2);
            break;
        case '*':
            // Multiplication
            printf("Resultat : %d\n", num1 * num2);
            break;
        case '/':
            // Division avec vérification si le second nombre n'est pas zéro pour éviter la division par zéro
            if (num2 != 0) {
                printf("Resultat : %d\n", num1 / num2);
            } else {
                printf("Erreur : Division par zero.\n");
            }
            break;
        case '%':
            // Modulo (reste de la division entière) avec vérification de division par zéro
            if (num2 != 0) {
                printf("Resultat : %d\n", num1 % num2);
            } else {
                printf("Erreur : Division par zero.\n");
            }
            break;
        case '&':
            // Opérateur AND bit à bit
            printf("Resultat : %d\n", num1 & num2);
            break;
        case '|':
            // Opérateur OR bit à bit
            printf("Resultat : %d\n", num1 | num2);
            break;
        case '~':
            // Opérateur NOT bit à bit
            printf("Resultat : %d\n", ~num1);
            break;
        default:
            // Message d'erreur si l'utilisateur entre un opérateur non reconnu
            printf("Operateur non reconnu.\n");
            break;
    }

    return 0;
}
